rpa-mapper/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/DefendantJsonMapper.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.cmc.rpa.mapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.otherparty.OrganisationDetails;
+import uk.gov.hmcts.cmc.domain.models.otherparty.SoleTraderDetails;
+import uk.gov.hmcts.cmc.domain.models.otherparty.TheirDetails;
+import uk.gov.hmcts.cmc.domain.models.party.HasContactPerson;
+import uk.gov.hmcts.cmc.domain.models.party.Party;
+import uk.gov.hmcts.cmc.domain.models.party.SoleTrader;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+import uk.gov.hmcts.cmc.rpa.mapper.helper.RPAMapperHelper;
+import uk.gov.hmcts.cmc.rpa.mapper.json.NullAwareJsonObjectBuilder;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObjectBuilder;
+
+import static uk.gov.hmcts.cmc.rpa.mapper.helper.Extractor.extractOptionalFromSubclass;
+
+@Component
+@SuppressWarnings({"LineLength"})
+public class DefendantJsonMapper {
+
+    @Autowired
+    private final AddressJsonMapper addressMapper;
+
+    public DefendantJsonMapper(AddressJsonMapper addressMapper) {
+        this.addressMapper = addressMapper;
+    }
+
+    public JsonArray map(TheirDetails defendant) {
+        JsonObjectBuilder jsonObjectBuilder = new NullAwareJsonObjectBuilder()
+            .add("type", defendant.getClass().getSimpleName().replace("Details", ""))
+            .add("name", defendant.getName())
+            .add("address", addressMapper.map(defendant.getAddress()))
+            .add("correspondenceAddress", defendant.getServiceAddress().map(addressMapper::map).orElse(null))
+            .add("emailAddress", defendant.getEmail().orElse(null))
+            .add("businessName", extractOptionalFromSubclass(defendant, SoleTraderDetails.class, value -> value.getBusinessName().map(RPAMapperHelper::prependWithTradingAs)))
+            .add("contactPerson", extractOptionalFromSubclass(defendant, HasContactPerson.class, HasContactPerson::getContactPerson))
+            .add("companiesHouseNumber", extractOptionalFromSubclass(defendant, OrganisationDetails.class, OrganisationDetails::getCompaniesHouseNumber));
+
+        return Json.createArrayBuilder().add(jsonObjectBuilder).build();
+    }
+
+    public JsonArray map(Party ownParty, TheirDetails oppositeParty, String defendantsEmail) {
```

Comment:
1. How about? `Party defendantFromClaim, TheirDetails defendantFromResponse`
The way it currently is I was a bit confused, also its not terribly obvious which method should be called in this class

2. Can the code be extracted so that there isn't 2 copies of it here

REQ

Reply:
1. Refactored 2. I rather leave it as because two method serve different jsons. Code is more maintainable separate as some of values are mapped different in this case.